### PR TITLE
Logging improvements

### PR DESCRIPTION
**Pull request by krazeelazy**
### I have:
- [X] [Searched](https://github.com/BoopChat/boop/pulls) the bugtracker for similar pull requests
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

## How Has This Been Tested?

I started the server on my local machine using `npm run dev-quick-restart` all expected messages were output to both the console and the log file. 

---

### What is the purpose of your *pull request*?
- [X] Improvement

### Description of your *pull request* and other information
 - Added new setup method for logger so we can determine when we are doing initial server setup and log an appropriate message:
     - `Starting server setup...` 
     - `Using <production or development> logger`
 -  Added log messages so we know what step of the server setup process we are on:
     - `Starting login service configuration`
     - `Configured <list of configured login services> as login services`
     - `Setting up server event listener`
     - `Boop server listening on port <port number>`

## Screenshots (if appropriate):
Screen Shot 2022-04-15 at 11 33 21 AM

### CODE REVIEW

File: server/bin/www [hunk 1]
```diff
@@ -97,21 +97,34 @@ function onError(error) {
...
     const bind = typeof addr === "string" ? "pipe " + addr : "port " + addr.port;
-    console.log("Boop server listening on " + bind);
+    logger.info("Boop server listening on " + bind);
 }
 
 /**
  * Gets the configured login services
  */
-function getLoginService () {
+async function getLoginService () {
```
Comment (JChris246) [inline server/bin/www:109]: Is async necessary for this function?

Comment (krazeelazy) [reply server/bin/www:109]: No. This is a remnant of testing different methodologies while trying to ensure the order of the messages. Will remove it.

File: server/bin/www.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    if (providerNames.length === 0) {
+        logger.error("Atleast one OAuth login service has to be configured");
+        process.exit(1);
+    } else if (providerNames.length >= 1) {
+        logger.info(
+            `Configured ${providerNames.join(", ")} ` +
+            `as${(providerNames.length === 1) ? " a" : ""} ` +
+            `login service${(providerNames.length > 1) && "s"}`
```
Comment (JChris246) [inline server/bin/www.js:133]: This is a logic error. Consider what the return value of this expression would be in both cases.

**Review by JChris246 — CHANGES_REQUESTED** [review]

File: server/bin/www.js (hunk 2 above)
Comment (krazeelazy) [reply server/bin/www.js]: Fixed it (4th message below)
Screen Shot 2022-04-15 at 7 05 38 PM

**Review by JChris246 — APPROVED** [review]
👍

**Pull request merged**